src/Controls/src/Core/TabIndexExtensions.cs

```diff
@@ -18,7 +18,7 @@ public static class TabIndexExtensions
 			countChildrenWithTabStopWithoutThis = 0;
 
 			Element parentPage = (element as Element)?.Parent;
-			while (parentPage != null && !(parentPage is Page))
+			while (parentPage is not Page)
```

Comment:
Does this actually do the same thing as before?

```csharp
parentPage != null && !(parentPage is Page)
```
would be `false` when `parentPage` is `null`

```csharp
parentPage is not Page
```
Would this still be false is `parentPage` is `null`?

Reply:
Hi Oliver, thanks for the question.

It would still be false if `parentPage` is `null`.

A simple example:

```
string s = null;
if (s is not string)
{
    Console.WriteLine("It's NOT a string.");
}
else
{
    Console.WriteLine("It's a string.");
}
```

prints `It's NOT a string.` to the console.

Reply:
I understand what the `is not` is doing, but the code you're replacing is not quite doing that. It's checking if it is not a `Page` _and_ is also not null:

```csharp
		object myString = "Hello";
		var result1 = (myString != null && !(myString is string));
		
		myString = null;
		var result2 = (myString is not string);	
		
		System.Console.WriteLine(result1);
		System.Console.WriteLine(result2);
```

`result1` is false
`result2` is true

Reply:
Thanks for this! I reverted to the previous `!= null && !(...)` pattern